clientv3/balancer.go

```diff
@@ -354,3 +363,47 @@ func getHost(ep string) string {
 	}
 	return url.Host
 }
+
+func (b *simpleBalancer) switchEndpoint() {
+	tNow := time.Now()
+	b.mu.RLock()
+
+	addrs := make([]grpc.Address, 0, len(b.addrs))
+	lastSwitchTime := b.lastSwitchTime
+
+	b.mu.RUnlock()
+
+	if lastSwitchTime.Add(30 * time.Second).After(tNow) {
+		// avoid too frequently switch
+		return
+	}
+
+
+	b.mu.Lock()
+	prevPinAddr := b.pinAddr
+	prevPinAddrEp := b.host2ep[prevPinAddr]
+	delete(b.host2ep, prevPinAddr)
+	for _, addr := range b.addrs {
+		if prevPinAddr != addr.Addr {
+			addrs = append(addrs, addr)
+		}
+
+	}
+	b.addrs = addrs
+	b.mu.Unlock()
+
+	select {
+	case b.updateAddrsC <- struct{}{}:
+	case <-b.stopc:
+	}
+
+
+	b.mu.Lock()
```

Comment:
this will deadlock because b.mu.Lock isn't released on the Wait; anything that tries to update pinAddrCond will wait forever to acquire the lock